File: app/src/main/java/com/kickstarter/models/Video.java
package com.kickstarter.models;

import android.os.Parcelable;
import android.support.annotation.Nullable;

import com.kickstarter.libs.qualifiers.AutoGson;

import auto.parcel.AutoParcel;

@AutoGson
@AutoParcel
public abstract class Video implements Parcelable {
  public abstract String base();
  public abstract String frame();
  public abstract String high();
  public abstract @Nullable String hls();
  public abstract @Nullable String webm();

  @AutoParcel.Builder
  public abstract static class Builder {
    public abstract Builder base(String __);
    public abstract Builder frame(String __);
    public abstract Builder high(String __);
    public abstract Builder hls(String __);
    public abstract Builder webm(String __);
    public abstract Video build();
  }

  public static Builder builder() {
    return new AutoParcel_Video.Builder();
  }

  public abstract Builder toBuilder();
}

File: app/src/main/java/com/kickstarter/models/Video.kt
package com.kickstarter.models

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

@Parcelize
class Video private constructor(
    private val base: String?,
    private val frame: String?,
    private val high: String?,
    private val hls: String?,
    private val webm: String?
) : Parcelable {
    fun base() = this.base
    fun frame() = this.frame
    fun high() = this.high
    fun hls() = this.hls
    fun webm() = this.webm

    @Parcelize
    data class Builder(
        private var base: String? = null,
        private var frame: String? = null,
        private var high: String? = null,
        private var hls: String? = null,
        private var webm: String? = null,
    ) : Parcelable {
        fun base(base: String?) = apply { this.base = base }
        fun frame(frame: String?) = apply { this.frame = frame }
        fun high(high: String?) = apply { this.high = high }
        fun hls(hls: String?) = apply { this.hls = hls }
        fun webm(webm: String?) = apply { this.webm = webm }

        fun build() = Video(
            base = base,
            frame = frame,
            high = high,
            hls = hls,
            webm = webm

        )
    }

    fun toBuilder() = Builder(
        base = base,
        frame = frame,
        high = high,
        hls = hls,
        webm = webm
    )

    override fun equals(obj: Any?): Boolean {
        var equals = super.equals(obj)
        if (obj is Video) {
            equals = base() == obj.base() &&
                frame() == obj.frame() &&
                high() == obj.high() &&
                hls() == obj.hls() &&
                webm() == obj.webm()
        }
        return equals
    }

    companion object {
        @JvmStatic
        fun builder() = Builder()
    }
}
